Algorithms/DataCompression/huffman.cs

```diff
@@ -25,186 +33,237 @@ public Huff(Huff leftChild, Huff rightChild)
             {
-                array[i] = d1[i];
+                huffmanObjects.Add(new Huff(fCharArray[i].ToString(), fFreqArray[i]));
             }
 
-            // fills the list 'list' with Huff-objects
-            for (var i = 0; i < pos; i++)
-            {
-                list.Add(new Huff(c1[i].ToString(), array[i]));
-            }
-            var stack = GetSortedStack(list);
-            while (stack.Count > 1)
-            {
-                var leftChild = stack.Pop();
-                var rightChild = stack.Pop();
-                var parentNode = new Huff(leftChild, rightChild);
-                stack.Push(parentNode);
-                stack = GetSortedStack(stack.ToList<Huff>());
-            }
-
-            // generated huffman tree
-            var parentNode1 = stack.Pop();
-            var man = new Man();
-
-            // generates and displays the huffman code
-            Console.WriteLine("\nHuffman Code:");
-            GenerateCode(parentNode1, "", man);
-            var cStr = " ";
-            foreach (var item in str)
-            {
-                var index = man.Data.IndexOf(item.ToString());
-                cStr += man.Codec.ElementAt(index);
-            }
-            var hfBits = 0;
-            foreach (var item in cStr.Replace(" ", ""))
-            {
-                hfBits++;
-            }
-            Console.WriteLine("Huffman Bits: " + hfBits);
-            Console.WriteLine("\nCoded String: ");
-            Console.WriteLine(cStr.Replace(" ", ""));
-            Console.ReadKey();
+            return huffmanObjects;
         }
-        public static Stack<Huff> GetSortedStack(IList<Huff> list)
+
+        private static Stack<Huff> GetSortedStack(IList<Huff> list)
         {
             for (var i = 0; i < list.Count; i++)
             {
                 for (var j = i + 1; j < list.Count; j++)
                 {
-                    if (list[i].Frequency > list[j].Frequency)
+                    if (list[i].Frequency <= list[j].Frequency)
                     {
-                        var tempNode = list[j];
-                        list[j] = list[i];
-                        list[i] = tempNode;
+                        continue;
                     }
+
+                    var tempNode = list[j];
+                    list[j] = list[i];
+                    list[i] = tempNode;
                 }
             }
+
             var stack = new Stack<Huff>();
-            for (var j = 0; j < list.Count; j++)
+            foreach (var t in list)
             {
-                stack.Push(list[j]);
+                stack.Push(t);
             }
 
             return stack;
         }
-        public static void GenerateCode(Huff parentNode, string code, Man man)
+
+        private static void GenerateCode(Huff parentNode, string code, Man man)
         {
-            if (parentNode != null)
+            var sbl = new StringBuilder(code);
+
+            while (parentNode != null)
```

Comment:
```suggestion
            var node = parentNode;
            while (parentNode != null)
```
Please introduce local variable and work with it instead of parentNode